src/bci_build/package/__init__.py

```diff
@@ -458,6 +458,12 @@ class BaseContainerImage(abc.ABC):
         default_factory=dict
     )
 
+    #: build flavors to produce for this container variant
+    build_flavor: str | None = None
+
+    #: Provide all the build flavors that exist, which is used for _multibuild/_service writing
+    all_build_flavors: list[str] | None = None
+
```

Comment:
This could be one single list IMHO. For example, if you create a tuple, dict, or a Flavor class.

```python
build_flavors: list[Flavor] = [Flavor(name="foo", current=True), Flavor(name="bar", current=False)]
```

If you have one item, you can remove some new checks in `__post_init__`.

Reply:
As originally indicated I wanted to make the all_build_flavor generation automatic, I just didn't have the time yesterday. 

did so now in the updated version.